### PR TITLE
Added web launch

### PR DESCRIPTION
**Pull request by cdavidjean**
- Added web launch
- Fixed some pub serve launch issues
- Fixed some console related issues introduced by the new DockViews.

### CODE REVIEW

**Comment by cdavidjean** [comment]
Note that: https://github.com/dart-atom/atom.dart/pull/52 has to be merged in first.

File: lib/launch/launch_web.dart [hunk 1]
```diff
@@ -13,9 +24,54 @@ class WebLaunchType extends LaunchType {
...
-    // TODO:
-    return new Future.error(new UnimplementedError());
+    Browser browser = deps[BrowserManager].browser;
+    if (browser == null) {
+      atom.notifications.addWarning('No browser configured.');
+      return new Future.value();
+    }
+
+    // Find pub serve for 'me'.
+    ServeLaunch pubServe = manager.launches.firstWhere((l) =>
```
Comment (devoncarew) [inline lib/launch/launch_web.dart]: Another option is to always start and stop a pub serve instance for a launch - that the `pub serve` instance has the same lifetime as the launch.

File: lib/plugin.dart [hunk 2]
```diff
@@ -299,6 +302,15 @@ class AtomDartPackage extends AtomPackage {
         'order': 1
       },
 
+      // debugger
+      'debugBrowserLocation': {
```
Comment (devoncarew) [inline lib/plugin.dart]: Perhaps just `browserLocation`?

File: lib/plugin.dart [hunk 3]
```diff
@@ -299,6 +302,15 @@ class AtomDartPackage extends AtomPackage {
...
+        'title': 'Debug Browser',
```
Comment (devoncarew) [inline lib/plugin.dart]: Chrome Location? Browser Location?

File: lib/plugin.dart [hunk 4]
```diff
@@ -299,6 +302,15 @@ class AtomDartPackage extends AtomPackage {
...
+        'description': 'The location of a Browser with Chrome DevTools Protocol.',
```
Comment (devoncarew) [inline lib/plugin.dart]: You may just want to call out Chrome specifically.

File: lib/browser.dart [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    return runner.onExit.then((_) => buf.toString());
+  }
+
+  List<String> execArgsFromYaml(yamlArgs) {
+    List execArgs = [];
+    if (yamlArgs is Map) {
+      yamlArgs.forEach((k, v) {
+        if (singletonBoolParameters.contains(k)) {
+          if (v) execArgs.add('--$k');
```
Comment (devoncarew) [inline lib/browser.dart]: do we need to check here if `v` is a `bool`?

File: lib/launch/launch_web.dart (hunk 1 above)
Comment (devoncarew) [inline lib/launch/launch_web.dart]: In that setup, the `serve` output would be just part of the launches output, and user's wouldn't have to know to start the `pub serve` before the web launch.

Comment (cdavidjean) [inline lib/launch/launch_web.dart]: You could also debug more than one html from the same project, having one pub serve for that use case is nice.
I'll add that as an option once the debugger works.  For now it's easier to build small steps.

File: lib/plugin.dart (hunk 2 above)
Comment (cdavidjean) [inline lib/plugin.dart]: Done.

File: lib/plugin.dart (hunk 3 above)
Comment (cdavidjean) [inline lib/plugin.dart]: Done.

File: lib/plugin.dart (hunk 4 above)
Comment (cdavidjean) [inline lib/plugin.dart]: The debug protocol seems to be used by more than one browser, according to:
https://github.com/cyrus-and/chrome-remote-interface#implementations
I'm happy either way.

File: lib/browser.dart (hunk 5 above)
Comment (cdavidjean) [inline lib/browser.dart]: AFAICS the Yaml parser seems to turn true and false into boolean.

File: lib/plugin.dart (hunk 4 above)
Comment (devoncarew) [inline lib/plugin.dart]: Hmm, perhaps something like:

```
The path to Chrome (or other browser supporting the Chrome DevTools protocol).
```

Comment (cdavidjean) [inline lib/plugin.dart]: Perfect.  Done.

**Pull request merged**